src/Uno.Toolkit.UI/Behaviors/SelectorExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.Logging;
+using Windows.System;
+using Windows.UI.ViewManagement;
+using Windows.Foundation.Collections;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using PipsPager = Microsoft.UI.Xaml.Controls.PipsPager;
+#endif
+
+namespace Uno.Toolkit.UI;
+public static partial class SelectorExtensions
+{
+	/// <summary>
+	/// Backing property for the <see cref="PipsPager"/> that will be linked to the desired <see cref="Selector"/> control.
+	/// </summary>
+	public static DependencyProperty PipsPagerProperty { get; } =
+		DependencyProperty.RegisterAttached("PipsPager", typeof(PipsPager), typeof(SelectorExtensions), new PropertyMetadata(null, OnPipsPagerChanged));
+
+	static void OnPipsPagerChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+	{
+		if (args.NewValue == args.OldValue || dependencyObject is not Selector flipView)
+			return;
+
+		var pipsPager = (PipsPager)args.NewValue;
+
+		var selectedIndexBinding = new Binding
+		{
+			Mode = BindingMode.TwoWay,
+			Source = flipView,
+			Path = new PropertyPath(nameof(flipView.SelectedIndex))
+		};
+
+		pipsPager.SetBinding(PipsPager.SelectedPageIndexProperty, selectedIndexBinding);
+
+
+		flipView.Items.VectorChanged -= OnItemsVectorChanged;
+		flipView.Items.VectorChanged += OnItemsVectorChanged;
+
+		pipsPager.NumberOfPages = flipView.Items.Count;
+
+		void OnItemsVectorChanged(IObservableVector<object> sender, IVectorChangedEventArgs @event) =>
+			pipsPager.NumberOfPages = flipView.Items.Count;
+	}
```

Comment:
We'll want to hook up to the Unloaded event on PipsPager and unsubscribe from all of the events from there. Similar to something like [here](https://github.com/unoplatform/uno.toolkit.ui/blob/62914f3607c877e5ad0589c5b17d045c10527119/src/Uno.Toolkit.UI/Controls/TabBar/TabBarSelectionIndicatorPresenter.cs#L62-L77).

Reply:
If I use the `SerialDisposable` is it still needed?

Reply:
I had an insight from the `GoNext` part, because this is static, so we can't use the `SerialDisposable`, because it will be attached to last control that set this DependencyProperty (DP). Does it make sense for you?